slamd/discovery/processing/algorithms/plot_generator.py

```diff
@@ -34,19 +34,33 @@ def create_target_scatter_plot(cls, plot_df):
         return json.dumps(fig, cls=plotly.utils.PlotlyJSONEncoder)
 
     @classmethod
-    def create_tsne_input_space_plot(cls, features_df, candidate_or_target):
-        tsne = TSNE(n_components=2, verbose=1, perplexity=40, n_iter=300, random_state=1000)
-        tsne_result = tsne.fit_transform(features_df)
+    def create_tsne_input_space_plot(cls, plot_df):
+        # The perplexity cannot be bigger than the number of data points (the length of the dataframe).
+        # Prevent this error case by picking the smallest of the two.
+        tsne = TSNE(n_components=2, verbose=1, perplexity=min(40, len(plot_df)), n_iter=300, random_state=1000)
+        # Exclude the columns that do not belong to the features
+        tsne_result = tsne.fit_transform(plot_df.drop(columns=['Row number', 'Utility', 'is_train_data']))
         tsne_result_df = pd.DataFrame(
-            {'tsne_1': tsne_result[:, 0],
+            {'Row number': plot_df['Row number'],
+             'tsne_1': tsne_result[:, 0],
              'tsne_2': tsne_result[:, 1],
-             'candidate_or_target': candidate_or_target}
+             'Utility': plot_df['Utility'],
+             'is_train_data': plot_df['is_train_data']}
         )
-
-        fig = px.scatter(tsne_result_df, x='tsne_1', y='tsne_2', color='candidate_or_target',
-                         title='Materials data in t-SNE coordinates: candidates and targets')
+        fig = px.scatter(tsne_result_df, x='tsne_1', y='tsne_2', color='Utility',
+                         symbol='is_train_data', custom_data=['Row number'],
```

Comment:
Personally I think diamonds aren't distinctive enough compared to dots. Would it be possible to use crosses?

Reply:
I just talked with Christoph and he mentioned that it would be to have the diamonds shown a little bit bigger than the dots.